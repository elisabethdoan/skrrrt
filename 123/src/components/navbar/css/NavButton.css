.Button {
    background: none;
    border: none;
    font-size: 30px;
    font-family: inherit;
    margin: 1rem 2rem 1rem 2rem;
}

.Button:hover {
    cursor: pointer;
    text-decoration: underline;
    text-decoration-color: #554488;
}

.ButtonSelected {
    background: none;
    border: none;
    font-size: 30px;
    font-family: inherit;
    text-decoration: underline;
    text-decoration-color: #554488;
    margin: 1rem 2rem 1rem 2rem;
}

@media only screen and (max-width: 600px) {
    .Button, .ButtonSelected {
        font-size: 25px;
        margin: 0.8rem 1.6rem 0.8rem 1.6rem;
    }
}
@media only screen and (max-width: 500px) {
    .Button, .ButtonSelected {
        font-size: 20px;
        margin: 0.6rem 1.2rem 0.6rem 1.2rem;
    }
}
@media only screen and (max-width: 400px) {
    .Button, .ButtonSelected {
        font-size: 15px;
        margin: 0.4rem 0.8rem 0.4rem 0.8rem;
    }
}
