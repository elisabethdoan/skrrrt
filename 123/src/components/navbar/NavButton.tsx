import React from 'react';
import { useHistory } from 'react-router-dom';
import './css/NavButton.css';

type buttonProps = {
  path: string,
  pathname: string,
  location: any
};

function NavButton({ path, pathname, location }: buttonProps) {
  const history = useHistory();

  const goToPath = () => {
    history.push(path);
  };

  const isSelected = (location === path);

  if (isSelected) {
    return (
      <button
        type="button"
        className="ButtonSelected"
        onClick={goToPath}
        data-testid="buttonSelected"
      >
        {pathname}
      </button>
    );
  }
  return (
    <button
      type="button"
      className="Button"
      onClick={goToPath}
      data-testid="button"
    >
      {pathname}
    </button>
  );
}

export default NavButton;
