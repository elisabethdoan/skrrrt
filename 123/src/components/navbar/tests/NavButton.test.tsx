import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import NavButton from '../NavButton';

describe('Navigation button', () => {
    
  afterEach(cleanup)

  test('Renders without crashing', () => {
    render(<NavButton path={'/'} pathname={'Home page'} location={'/'}/>)
  });

  test('Renders correct when selected', () => {
    render(<NavButton path={'/'} pathname={'Home page'} location={'/'}/>)
    expect(screen.queryByTestId('buttonSelected')).toBeTruthy()
    expect(screen.queryByTestId('button')).toBeFalsy()
  });

  test('Renders correct when selected', () => {
    render(<NavButton path={'/'} pathname={'Home page'} location={'/item1'}/>)
    expect(screen.queryByTestId('buttonSelected')).toBeFalsy()
    expect(screen.queryByTestId('button')).toBeTruthy()
  });

  test('Renders with correctly', () => {
    render(<NavButton path={'/'} pathname={'Home page'} location={'/'}/>)
    expect(screen.queryByTestId('buttonSelected')).toMatchSnapshot()
  })

});