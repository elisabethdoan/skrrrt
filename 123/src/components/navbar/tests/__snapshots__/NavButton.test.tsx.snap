// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation button Renders with correctly 1`] = `
<button
  class="ButtonSelected"
  data-testid="buttonSelected"
  type="button"
>
  Home page
</button>
`;
